packages/protocol/src/protocol/DeviceCommissioner.ts

```diff
@@ -0,0 +1,280 @@
+        }
+        this.#enterCommissioningMode(this.#windowStatus, this.#activeDiscriminator).catch(error =>
+            logger.warn("Error sending announcement:", error),
+        );
+    }
+
+    async #enterCommissioningMode(
+        windowStatus: AdministratorCommissioning.CommissioningWindowStatus,
+        discriminator?: number,
+    ) {
+        this.#windowStatus = windowStatus;
+        const commissioningConfig = this.#context.commissioningConfig.values;
+        await this.#context.advertiser.enterCommissioningMode(
+            windowStatus === AdministratorCommissioning.CommissioningWindowStatus.EnhancedWindowOpen
+                ? CommissioningMode.Enhanced
+                : CommissioningMode.Basic,
+            {
+                ...commissioningConfig.productDescription,
+                discriminator: discriminator ?? commissioningConfig.discriminator,
+            },
+        );
+    }
+
+    async #becomeCommissionable(
+        windowStatus: AdministratorCommissioning.CommissioningWindowStatus,
+        activeCommissioningEndCallback?: () => void,
+        discriminator?: number,
+    ) {
+        if (
+            this.#windowStatus === windowStatus &&
+            (discriminator === undefined || discriminator === this.#activeDiscriminator)
+        ) {
+            // We want to re-announce
+            return this.reactivateAdvertiser();
+        }
+        if (this.#windowStatus !== AdministratorCommissioning.CommissioningWindowStatus.WindowNotOpen) {
+            throw new InternalError(`Commissioning window already open with different mode (${this.#windowStatus})!`);
+        }
+        if (this.#activeCommissioningEndCallback !== undefined) {
+            throw new InternalError("Commissioning window already open with different callback!");
+        }
+        this.#activeCommissioningEndCallback = activeCommissioningEndCallback;
+        this.#activeDiscriminator = discriminator;
+
+        // MDNS is sent in parallel
+        // TODO - untracked promise
+        this.#enterCommissioningMode(windowStatus, discriminator).catch(error =>
+            logger.warn("Error sending announcement:", error),
+        );
+    }
+
+    async endCommissioning() {
+        if (this.#windowStatus === AdministratorCommissioning.CommissioningWindowStatus.WindowNotOpen) {
+            return;
+        }
+
+        logger.debug("Commissioning mode ended, stop announcements.");
+        // Remove PASE responder when we close enhanced commissioning window or node is commissioned
+        if (
+            this.#windowStatus === AdministratorCommissioning.CommissioningWindowStatus.EnhancedWindowOpen ||
+            this.#context.fabrics.getFabrics().length > 0
+        ) {
+            this.#context.secureChannelProtocol.removePaseCommissioner();
+        }
+        this.#windowStatus = AdministratorCommissioning.CommissioningWindowStatus.WindowNotOpen;
+
+        await this.#context.advertiser.exitCommissioningMode();
```

Comment:
I thuink if you move this line below the callback handling block the error is gone ... the question is still why this consistently fails/blocks, but the "old" code has these two blocks exchanged

Reply:
Ah OK, good catch.  Will try it